### PR TITLE
[docs] AnimateDiff FreeNoise 

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
# What does this PR do?

Adds documentation for AnimateDiff FreeNoise introduced in #9262

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu, @sayakpaul

### CODE REVIEW

File: docs/source/en/api/pipelines/animatediff.md [hunk 1]
```diff
@@ -914,6 +914,79 @@ export_to_gif(frames, "animatelcm-motion-lora.gif")
...
+# Can be a single prompt, or a dictionary with frame timesteps
+prompt = {
+    0: "A caterpillar on a leaf, high quality, photorealistic",
+    40: "A caterpillar transforming into a cocoon, on a leaf, near flowers, photorealistic",
+    80: "A cocoon on a leaf, flowers in the backgrond, photorealistic",
+    120: "A cocoon maturing and a butterfly being born, flowers and leaves visible in the background, photorealistic",
+    160: "A beautiful butterfly, vibrant colors, sitting on a leaf, flowers in the background, photorealistic",
+    200: "A beautiful butterfly, flying away in a forest, photorealistic",
+    240: "A cyberpunk butterfly, neon lights, glowing",
+}
```
Comment (sayakpaul) [inline docs/source/en/api/pipelines/animatediff.md:960]: The keys denote what? I would maybe clarify that a bit.

**Review by sayakpaul — APPROVED** [review]
Just two minor comments otherwise it's looking lit!

File: docs/source/en/api/pipelines/animatediff.md [hunk 2]
```diff
@@ -914,6 +914,89 @@ export_to_gif(frames, "animatelcm-motion-lora.gif")
...
+- [AnimateDiffVideoToVideoPipeline]
+- [AnimateDiffVideoToVideoControlNetPipeline]
+
+In order to use FreeNoise, a single line needs to be added to the inference code after loading your pipelines.
+
+```diff
++ pipe.enable_free_noise()
+```
+
+After this, either a single prompt could be used, or multiple prompts can be passed as a dictionary of integer-string pairs. The integer keys of the dictionary correspond to the frame index at which the influence of that prompt would be maximum. Each frame index should map to a single string prompt. The prompts for intermediate frame indices, that are not passed in the dictionary, are created by interpolating between the frame prompts that are passed. By default, simple linear interpolation is used however one can customize this behaviour by a callback to the `prompt_interpolation_callback` parameter when enabling FreeNoise.
```
Comment (stevhliu) [inline docs/source/en/api/pipelines/animatediff.md:935]:
```suggestion
After this, either a single prompt could be used, or multiple prompts can be passed as a dictionary of integer-string pairs. The integer keys of the dictionary correspond to the frame index at which the influence of that prompt would be maximum. Each frame index should map to a single string prompt. The prompts for intermediate frame indices, that are not passed in the dictionary, are created by interpolating between the frame prompts that are passed. By default, simple linear interpolation is used. However, you can customize this behaviour with a callback to the `prompt_interpolation_callback` parameter when enabling FreeNoise.
```

File: docs/source/en/api/pipelines/animatediff.md [hunk 3]
```diff
@@ -914,6 +914,89 @@ export_to_gif(frames, "animatelcm-motion-lora.gif")
...
+    num_inference_steps=10,
+    generator=torch.Generator("cpu").manual_seed(0),
+)
+
+# Save video
+frames = output.frames[0]
+export_to_video(frames, "output.mp4", fps=16)
+```
+
+#### FreeNoise memory savings
```
Comment (stevhliu) [inline docs/source/en/api/pipelines/animatediff.md:988]:
```suggestion
### FreeNoise memory savings
```

**Review by stevhliu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**